src/m365/commands/login.ts

```diff
@@ -163,46 +163,48 @@ class LoginCommand extends Command {
           break;
       }
 
-      auth
-        .ensureAccessToken(auth.defaultResource, logger, this.debug)
-        .then((): void => {
-          auth.service.connected = true;
-          cb();
-        }, (rej: Error): void => {
-          if (this.debug) {
-            logger.logToStderr('Error:');
-            logger.logToStderr(rej);
-            logger.logToStderr('');
-          }
-
-          cb(new CommandError(rej.message));
-        });
-    };
+      try {
+        await auth.ensureAccessToken(auth.defaultResource, logger, this.debug);
+        auth.service.connected = true;
 
-    auth
-      .clearConnectionInfo()
-      .then((): void => {
-        logout();
-        login();
-      }, (error: any): void => {
+      }
+      catch(error: any) {
         if (this.debug) {
-          logger.logToStderr(new CommandError(error));
+          logger.logToStderr('Error:');
+          logger.logToStderr(error);
+          logger.logToStderr('');
         }
 
-        logout();
-        login();
-      });
+        throw new CommandError(error.message);
+      }
+    };
+
+    try {
+      await auth.clearConnectionInfo();
+    }
+    catch (error: any) {
+      if (this.debug) {
+        logger.logToStderr(new CommandError(error));
+      }
+
+      logout();
```

Comment:
Since we want to run login and logout either way, let's move them to a `finally` block which will run no matter if clearing connection info succeeded or failed. Then, we won't need to repeat it on line 194.

Reply:
Nice suggestion indeed, noticed that right now if the catch branch is executed, `login` and `logout` will be executed twice, which is not ideal. `Finally` block will fix this.